app/uk/gov/hmrc/incometaxsessiondata/controllers/SessionController.scala

```diff
@@ -28,28 +28,25 @@ import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton()
-class SessionController @Inject() (
-  cc: ControllerComponents,
-  authentication: AuthenticationPredicate,
-  sessionService: SessionService
-)(implicit ec: ExecutionContext)
-    extends BackendController(cc)
+class SessionController @Inject()(
+                                   cc: ControllerComponents,
+                                   authentication: AuthenticationPredicate,
+                                   sessionService: SessionService
+                                 )(implicit ec: ExecutionContext)
+  extends BackendController(cc)
     with Logging {
 
-  def getById(sessionID: String): Action[AnyContent] = authentication.async { request =>
+  def get(): Action[AnyContent] = authentication.async { request =>
     // Here is required internalID => request.internalId and request.sessionId
-    sessionService.get(sessionID) map {
-      case Right(Some(session: SessionData)) =>
-        logger.info(s"[SessionController][getById]: Successfully retrieved session data: $session")
+    sessionService.get(request) map {
+      case Some(session: SessionData) =>
+        logger.info(s"[SessionController][get]: Successfully retrieved session data: $session")
```

Comment:
I would suggest to log only: sessionId and internalId as they are part of the primary key/index, rest of the data we are encrypting, so we should not log them I think

Reply:
The Session has been converted to SessionData here, so we would only have sessionId, mtditid, nino and utr. Shall I just change it to sessionId in this case then?

Reply:
I am referring to what is permitted to log in general in this service, my understanding we should not log any `nino, utrl and mtdItId`
> https://confluence.tools.tax.service.gov.uk/display/DTRG/Logging+best+practices

Also please check header/title in the next document, it is an ultimate goal of this service: https://confluence.tools.tax.service.gov.uk/pages/viewpage.action?spaceKey=SASUB&title=Removal+of+personal+identifiable+information+from+session+cookie

You would also need to check other places where you log these identifiers